### ISSUE TITLE
program no response

### ISSUE THREAD

**Issue by r91085** [MeanEYE/Sunflower#293]
The program has no response when I right click mouse and select " Properties"

**Comment by MeanEYE** [comment]
Is it doing something? Can you see system monitor and if Sunflower has some CPU load?

**Comment by r91085** [comment]
Yes , Sunflower has cpu load , but  I don't know what is it doing now .
The UI has no response , and only I can do is force end it .

By the way , Is there a commands variables of "Name of the current directory of none-ctive list." or something like that I can get two panel's directorypath ?

**Comment by MeanEYE** [comment]
On what did you right click? Only thing that actually happens when you right click is menu position calculation which shouldn't really take much CPU time.

As for paths, you mean reference path in your command? If so, yes. [There is a way](https://github.com/MeanEYE/Sunflower/wiki/Commands-menu).

**Comment by r91085** [comment]
Oh ~ sunflower has no response after right click menu popup and I click the item "Properties" .
So I think it should be something wrong during launch function of "Properities"

I had already read the page of "your command" , but I can find the variable of " current directory path of inactive panel ". It is like "%d, %D - Name of the current directory of active list."
Mabye I miss something ?

**Comment by MeanEYE** [comment]
You are not missing anything. There's no replacement for opposite directory. Feel free to report this as feature request separately and I'll see that it's in next release.

**Comment by r91085** [comment]
So about this issue of no response , is there anything I can do to find what happen ?
If  Properties panel can be showed , I can not change it's permission setting

**Comment by MeanEYE** [comment]
I checked the code and there are no apparent reasons why this should happen. Can you provide me with more information about your system. Which desktop environment you are using, distribution, etc.

**Comment by MeanEYE** [comment]
More importantly which version of Sunflower you are using.

**Comment by r91085** [comment]
Sunflower is 0.3 (61)
linux , Debian rodete (Linux 4.9.0) on amd64
Display Manager is lightdm 

Is it possible enable some log function to see it stop on which-process?

**Comment by MeanEYE** [comment]
We don't gave logging of such kind built in the program itself. Can you explain in more detail what happens. Is there a new window shown or not?

**Comment by r91085** [comment]
No , after I click "Properties" , nothing happen , No new window .
And sunflower main window also suspend , only I can do is force ending it .....

**Comment by MeanEYE** [comment]
Can you start Sunflower from the terminal and provide any messages that are displayed.

**Comment by r91085** [comment]
I tried it .
But it did not output any message after  I click "Properties"  ......

**Comment by MeanEYE** [comment]
There's nothing special with properties window that is not used in other parts of the program. Do you experience similar behavior somewhere else in Sunflower?

**Comment by r91085** [comment]
No , other funciton of sunflower seems to work well .
I am also confused with this problem .

It seems sunflower waiting for some api return or it can not access some information in Properities panel like permission ? 

But other file manager does not has this kind of properties issue ....

**Comment by MeanEYE** [comment]
There's nothing special with properties window that is not used in other parts of the program. Do you experience similar behavior somewhere else in Sunflower?

**Comment by MeanEYE** [comment]
Few questions:

- Do you use emblems;
- Does this happen on any item type (directory / file);
- Does this happen in different directories?

**Comment by r91085** [comment]
1. I am sorry . I don't understand what is "emblems" . Could you please explain it for me ?
2. I tested it on both directory / file , and all of them have the same problem.
3. Yes , I think I had tried this function on more than ten folder .

**Comment by r91085** [comment]
I print log to trace .
I think I am sure it is into an infinite loop on below selection :

> 		if self._provider.is_local:
> 			for i, user in enumerate(pwd.getpwall()):
> 				self._list_owner.append((user.pw_name, user.pw_uid))
> 				if user.pw_uid == stat.user_id:
> 					self._combobox_owner.set_active(i)

**Comment by MeanEYE** [comment]
Infinite loop should be blocked by:
https://github.com/MeanEYE/Sunflower/blob/6061a14e2bbf39c9c65a4a24e1e45e53a1ff8f2c/application/gui/properties_window.py#L325-L326

Can you try commenting out code you've linked and see if the issue remains.

**Comment by r91085** [comment]
If I comment the code below : 

> 			for i, user in enumerate(pwd.getpwall()):
> 				print ("xxx _ownership_update 3.0"+user.pw_name)
> 				self._list_owner.append((user.pw_name, user.pw_uid))
> 				if user.pw_uid == stat.user_id:
> 					self._combobox_owner.set_active(i)
> 			print ("xxx _ownership_update 3.1")
> 			for i, group in enumerate(grp.getgrall()):
> 				print ("xxx _ownership_update 3.1"+user.pw_name)
> 				self._list_group.append((group.gr_name, group.gr_gid))
> 				if group.gr_gid == stat.group_id:
> 					self._combobox_group.set_active(i)

then the properties panel can be shown , but the information of Ownership in permission tab has no data...

**Comment by r91085** [comment]
I check the num of list .
There are 53xxxx items in pwd.getpwall() and 17xxx items in grp.getgrall() .
So I think mabye it is not an infinite loop , but too many items in it to process .

Is there anything I can do ?

**Comment by MeanEYE** [comment]
Okay, this issue never happened before but it makes sense that is happening. Problem is that handler blocking is not doing its job and that causes infinite loop. I will accept this issue and patch it soon. Sorry for the inconvenience it caused and thanks for debugging effort, it's much appreciated.

**Comment by MeanEYE** [comment]
Can you confirm if this issue is still present with current development version. Since it was reported we have switched to GTK3 and Python3. Issue should be resolved, but I'd like to have it confirmed.

**Comment by r91085** [comment]
I just see your reply . I will test it and let you know the result , thanks~

**Comment by MeanEYE** [comment]
Thanks. Looking forward to hearing from you.

**Comment by r91085** [comment]
Could you tell me how to use the development version ? I use the .deb file to install before , now got zip file on github , but after I run the Sunflower.desktop in it , it seems there is no Sunflower.py int the zip file ....

**Comment by MeanEYE** [comment]
Sorry about that. Unpack that zip file somewhere and then just open terminal and navigate to same directory. From there you can just do `python3 -m sunflower`.

**Comment by r91085** [comment]
This issue seem still exist in my computer ....

**Comment by MeanEYE** [comment]
Are you still running the same version of operating system?

**Comment by r91085** [comment]
I used the latest zip version from github .
